dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/netty/jfapchannel/NettyToWsBufferDecoder.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.sib.netty.jfapchannel;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.sib.jfapchannel.JFapChannelConstants;
+import com.ibm.ws.sib.jfapchannel.buffer.WsByteBufferPool;
+import com.ibm.ws.sib.utils.ras.SibTr;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.ByteToMessageDecoder;
+
+/**
+ * Netty decoder for transforming incoming ByteBuf objects to WsByteBuffer objects
+ * we work with in the JMS bundles.
+ * 
+ */
+public class NettyToWsBufferDecoder extends ByteToMessageDecoder {
+
+	/** Trace */
+	private static final TraceComponent tc = SibTr.register(NettyToWsBufferDecoder.class,
+			JFapChannelConstants.MSG_GROUP,
+			JFapChannelConstants.MSG_BUNDLE);
+
+	/** Log class info on load */
+	static
+	{
+		if (tc.isDebugEnabled())
+			SibTr.debug(tc,
+					"@(#) SIB/ws/code/sib.jfapchannel.client.rich.impl/src/com/ibm/ws/sib/netty/jfapchannel/NettyToWsBufferDecoder.java, SIB.comms, WASX.SIB, uu1215.01 1.1");
+	}
+
+
+	@Override
+	protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
+
+		if (tc.isEntryEnabled())
+			SibTr.entry(this, tc, "decode", ctx.channel());
+
+		if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+			SibTr.debug(this, tc, "decode", ctx.channel().remoteAddress() + " decoding message [ " + in.toString(StandardCharsets.UTF_8) + " ] from Netty ByteBuf to WSByteBuffer");
+		}
+
+		ByteBuf temp = in.readBytes(in.readableBytes());
+
+		byte[] bytes;
+		int offset;
+		int length = temp.readableBytes();
+
+		if (temp.hasArray()) {
```

Comment:
would we not know the answer to this statically - does the testing path ever go down both sides of this if?